backend/src/apiserver/config/config.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2025 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/client"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/common"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/resource"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/server"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"os"
+	"time"
+)
+
+// deprecated
+type deprecatedConfig struct {
```

Comment:
Why do we need a separate struct? The new struct looks backwards compatible.

Reply:
the separation to me makes it clear where the old one has left off, it handles very niche scenarios. For example if we deprecate "File" in the new struct, and eventually remove it before we remove support for the old config, then it will stop being backwards compatible. I doubt we ever hit this, however. So if you feel strongly about it I can remove it.

Reply:
I don't mind changing it, one thing to note is that by having it as a separate struct we also get the benefit of marking it // deprecated, which shows up on ide's (at least goland), making it easy to identify which code is leveraging this code: 



![image](https://github.com/user-attachments/assets/0684627c-1532-4ab8-b8ae-8f1c31d08d7b)

Reply:
Oh I see. I think we would want to stop supporting two different types of config files, otherwise this feels like tech debt we could accrue, and to do that we would have to mark this format deprecated, with a warning to the user to switch over to the new format as per the log message. No?

Reply:
@HumairAK but are they actually different types of config files? All the new fields are optional and the default values of the new fields result in the same behavior as before.

In other words, isn't the JSON representation of `deprecatedConfig` a valid `configPipelines`?

Reply:
I see what you mean. I've come around to your point of view. This can be resolved!